src/package_utils.rs

```diff
@@ -0,0 +1,70 @@
+use zed_extension_api::{self as zed, Result, serde_json};
+
+#[derive(Debug, serde::Deserialize)]
+struct Process {
+    platform: String,
+    arch: String,
+}
+
+fn get_node_process() -> Result<Process> {
+    let output = zed::Command {
+        command: "node".into(),
+        args: vec!["-p".into(), "JSON.stringify(process)".into()],
+        env: vec![],
+    }
+    .output()?;
+
+    if !output.status.is_some_and(|status| status == 0) {
+        return Err(format!("unable to get the node process variable"));
+    }
+
+    serde_json::from_slice::<Process>(output.stdout.as_slice()).map_err(|e| e.to_string())
+}
```

Comment:
This is what I am this most unsettled by. I assume its possible to get an equivalent value from rust and perform some mapping to the node values, however, I suspect the wasm target will put an end to that path.